lib/at_shell/at_shell.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+
+#include <shell/shell.h>
+#include <nrf_modem_at.h>
+#include <modem/at_monitor.h>
+
+static struct shell *global_shell;
+static const char at_usage_str[] =
+	"Usage: at <subcommand>\n"
+	"\n"
+	"Subcommands:\n"
+	"  events_enable     Enable AT event handler which prints AT notifications\n"
+	"  events_disable    Disable AT event handler\n"
+	"\n"
+	"Any other subcommand is interpreted as AT command and sent to the modem.\n";
+
+AT_MONITOR(at_shell_monitor, ANY, at_notif_handler, PAUSED);
+
+static void at_notif_handler(const char *response)
+{
+	shell_print(global_shell, "AT event handler:\n%s", response);
+}
+
+static void at_print_error_info(const struct shell *shell, int error)
```

Comment:
In `modem_shell`, I've done conversion from `at_cmd` library to `nrf_modem_at` few months ago and I did this similar function at first. Then I started to look at the response and it's actually pretty much the response coming from the modem so I ended up just printing modem response when err > 0. There are testing aspects why we could use `nrf_modem_at_err_type(error)` and `nrf_modem_at_err(error)` but other than that, the actual response from modem is really nice given it's formatted pretty much the way we want.

You can see MOSH PR on nrf_modem_at conversion now at:
https://github.com/nrfconnect/sdk-nrf/pull/6116

@tokangas, @jhirsi, Please also comment if you have a preference on which way to do it.

Reply:
Yes, modem response would be enough for me. Testing automation folks could be consulted as well.

I think that the additional feature that would add value would be to translate CME error values as text. There could one mapping api in NCS, although not sure if the at_shell is a correct place.

Reply:
Good points. Seems like just printing the response is good enough here, so I'll remove `at_print_error_info` and just print it.
I saw some translation of CME codes somewhere, I'll see if I can dig them up. Can probably add that in another PR.